ldap/servers/slapd/back-ldbm/db-mdb/mdb_layer.c

```diff
@@ -2300,11 +2300,21 @@ int dbmdb_cursor_get_recno(dbi_cursor_t *cursor, MDB_val *dbmdb_key, MDB_val *db
     MDB_cursor *newcur = NULL;
     int cmpres = 0;
     int rc = 0;
+    struct ldbminfo *li = (struct ldbminfo *)cursor->be->be_database->plg_private;
+    dbmdb_ctx_t *ctx = MDB_CONFIG(li);
+    dbmdb_dbi_t *dbi = &ctx->dbi_slots[mdb_cursor_dbi(cursor->cur)];
 
     rc = MDB_CURSOR_GET(cursor->cur, &curpos_key, &curpos_data, MDB_GET_CURRENT);
     if (rc != 0) {
         return rc;
     }
+    if (dbi->dbname[0] == '~') {
```

Comment:
Just to clarify (as I am not very proficient with this area), `'~'` is used as a special prefix for VLV purposes?

Reply:
Not really surprising you do not know about it as it is specific to lmdb vlv handling.
It is used for handling the vlv index cache that keep trace of the key that are at some position in the index file